### PR TITLE
Adds flag to clean up cilium state before startup

### PR DESCRIPTION
**Pull request by abhide15**

### CODE REVIEW

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed [hunk 1]
```diff
@@ -32,6 +32,22 @@ spec:
         prometheus.io/port: "9090"
     spec:
       serviceAccountName: cilium
+      initContainers:
+      - name: clean-cilium-state
+        image: busybox
+        command: ['sh', '-c', 'if [ ${CLEAN_CILIUM_STATE} ]; then rm -rf /var/run/cilium/state; rm -rf /sys/fs/bpf/tc/globals; fi']
```
Comment (joestringer) [inline examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: Should the `/sys/fs/bpf/tc/globals` rm only do it for `/sys/fs/bpf/tc/globals/cilium_*`?

**Review by joestringer — COMMENTED** [review]
Please sign off on your commits (You can do so using `git commit -s`, or to amend the current commit, `git commit --amend -s`). The meaning is described [here](http://docs.cilium.io/en/latest/contributing/#developer-s-certificate-of-origin).

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed (hunk 1 above)
Comment (joestringer) [reply examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: I guess if we want to catch all maps, we'd need also #3681 for the above comment to work.

**Comment by aanm** [comment]
test-me-please

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed (hunk 1 above)
Comment (abhide15) [reply examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: Fixed

**Comment by manalibhutiyani** [comment]
test-missed-k8s

**Review by joestringer — APPROVED** [review]
LGTM, I assume you've already manually tested this?

**Comment by joestringer** [comment]
test-me-please

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed [hunk 2]
```diff
@@ -32,6 +32,22 @@ spec:
...
+        volumeMounts:
+          - name: bpf-maps
+            mountPath: /sys/fs/bpf
+          - name: cilium-run
+            mountPath: /var/run/cilium
+        env:
+          - name: "CLEAN_CILIUM_STATE"
+            valueFrom:
+                configMapKeyRef:
+                  name: clean-cilium-state
```
Comment (aanm) [inline examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: Why not use the existing ConfigMap and add this option in there?

**Review by aanm — CHANGES_REQUESTED** [review]

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed (hunk 2 above)
Comment (abhide15) [reply examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: Fixed

File: examples/kubernetes/templates/v1/cilium-cm.yaml [hunk 3]
```diff
@@ -25,6 +25,8 @@ data:
   # If you want to run cilium in debug mode change this value to true
   debug: "false"
   disable-ipv4: "false"
+  # If you want to clean cilium state; uncomment the following lines
+  #clean-cilium-state: "true"
```
Comment (aanm) [inline examples/kubernetes/templates/v1/cilium-cm.yaml]: You can uncomment the line and have the value set to `false`, no?

**Review by aanm — CHANGES_REQUESTED** [review]

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed [hunk 4]
```diff
@@ -32,6 +32,22 @@ spec:
         prometheus.io/port: "9090"
     spec:
       serviceAccountName: cilium
+      initContainers:
+      - name: clean-cilium-state
+        image: busybox
+        command: ['sh', '-c', 'if [ ${CLEAN_CILIUM_STATE} ]; then rm -rf /var/run/cilium/state; rm -rf /sys/fs/bpf/tc/globals/cilium_*; fi']
```
Comment (rlenglet) [inline examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: If the option is set to `false` this doesn’t seem to do the right thing.
This should be:
```
... if [ “${CLEAN_CILIUM_STATE}” = “true” ] ...
```

**Review by rlenglet — CHANGES_REQUESTED** [review]

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed (hunk 4 above)
Comment (abhide15) [reply examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: yes.. fixing it

File: examples/kubernetes/templates/v1/cilium-cm.yaml (hunk 3 above)
Comment (abhide15) [reply examples/kubernetes/templates/v1/cilium-cm.yaml]: I was just checking if this ENV variable is present.. I will change this

File: examples/kubernetes/templates/v1/cilium-ds.yaml.sed (hunk 4 above)
Comment (rlenglet) [reply examples/kubernetes/templates/v1/cilium-ds.yaml.sed]: And if you wanted to check whether the variable was set, this test should have been different anyway:
```
if [ -n "${CLEAN_CILIUM_STATE}" ] ...
```

**Review by aanm — CHANGES_REQUESTED** [review]
One last thing is to regenerate the kubernetes spec files `make CILIUM_VERSION=latest`

**Comment by abhide15** [comment]
@aanm fixed your comment and added generated cilium k8s yaml files

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by abhide15** [comment]
test-me-please

**Comment by abhide15** [comment]
test-missed-k8s

**Review by manalibhutiyani — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

**Comment by abhide15** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by abhide15** [comment]
submitted another PR using a different branch. This new PR is merged. 
Closing this PR as its not required.

**Pull request closed without merging**